libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -493,24 +493,32 @@ bool ASTJsonConverter::visit(ModifierInvocation const& _node)
 bool ASTJsonConverter::visit(EventDefinition const& _node)
 {
 	m_inEvent = true;
-	setJsonNode(_node, "EventDefinition", {
+	std::vector<pair<string, Json::Value>> _attributes = {
 		make_pair("name", _node.name()),
 		make_pair("nameLocation", sourceLocationToString(_node.nameLocation())),
 		make_pair("documentation", _node.documentation() ? toJson(*_node.documentation()) : Json::nullValue),
 		make_pair("parameters", toJson(_node.parameterList())),
 		make_pair("anonymous", _node.isAnonymous())
-	});
+	};
+	if (m_stackState >= CompilerStack::State::AnalysisPerformed)
+			_attributes.emplace_back(make_pair("eventSelector", _node.functionType(true)->externalIdentifierHex()));
```

Comment:
```suggestion
			_attributes.emplace_back(make_pair("eventSelector", toHex(u256(h256::Arith(keccak256(_node.functionType(true)->externalSignature())))));
```
Something like this.